src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4encoder/HaivisionX4EncoderCommunicator.java

```diff
@@ -553,10 +605,36 @@ else if (EncodingProfile.MAIN_10.getName().equals(currentEncodingProfile)) {
 
 				AdvancedControllableProperty encodingProfileControlProperty = controlDropdown(extendedStatistics, encodingProfileDropdown, property, value);
 				addAdvanceControlProperties(advancedControllableProperties, encodingProfileControlProperty);
+
+				property = videoName + HaivisionConstant.HASH + VideoControllingMetric.CHROMA_SUBSAMPLING.getName();
+				String[] chromaSubsampling;
+				String chromaValue = extendedStatistics.get(property);
+				if (EncodingProfile.BASELINE.getName().equals(value) || EncodingProfile.MAIN.getName().equals(value) || EncodingProfile.HIGH.getName()
+						.equals(value)) {
+					chromaSubsampling = ChromaSubSampling.namesIsBaselineOrMainOrHigh();
+					chromaValue = ChromaSubSampling.BIT_420_8.getName();
+				}
+				//If Encoding profile is Main 10 or High 10, the Chroma subsampling list includes {4:2:0 8-bit; 4:2:0 10-bit}
+				else if (EncodingProfile.MAIN_10.getName().equals(value) || EncodingProfile.HIGH_10.getName().equals(value)) {
+					chromaSubsampling = ChromaSubSampling.namesIsMain10OrHigh10();
+					if (ChromaSubSampling.BIT_422_8.getName().equals(chromaValue)) {
+						chromaValue = ChromaSubSampling.BIT_420_8.getName();
+					}
+
+					if (ChromaSubSampling.BIT_422_10.getName().equals(chromaValue)) {
+						chromaValue = ChromaSubSampling.BIT_420_10.getName();
+					}
+				}
+				//If Encoding profile is Main 4:2:2 10 or High 4:2:2, the Chroma subsampling list includes {4:2:0 8-bit; 4:2:0 10-bit, 4:2:2 8-bit, 4:2:2 10-bit}
```

Comment:
not a big deal, but the format is a bit off because of the mid-blocks comment, maybe would look better if it is inside of the _else_ block :)

Reply:
I fixed it